package/src/core/define-integration.ts

```diff
@@ -43,7 +45,16 @@ export const defineIntegration = <
 	}) => ExtendedHooks<TPlugins>;
 }): ((options?: z.input<TOptionsSchema>) => AstroIntegration) => {
 	return (_options: z.input<TOptionsSchema> = {}) => {
-		const options = optionsSchema?.parse(_options) as z.output<TOptionsSchema>;
+		const parsedOptions = (optionsSchema || z.record(z.any())).safeParse(_options, { errorMap });
```

Comment:
May be a good idea to make the default explicit in the generic as well
I can't send a suggestion in that line because it wasn't change, but I believe it would become this now:

```patch
- TOptionsSchema extends import("astro/zod").AnyZodObject,
+ TOptionsSchema extends z.AnyZodObject = z.ZodRecord<z.ZodAny>,
```

Reply:
I tried this but `z.ZodRecord<z.ZodAny>`does not satisfy `z.AnyZodObject`
![image](https://github.com/florian-lefebvre/astro-integration-kit/assets/19967622/3e46ac4e-6073-4d7f-9660-de6ab6d57ec8)

Reply:
That also works. Weird that for Zod a record is not an object